### PR TITLE
Feature/5x breaking changes docs

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

File: docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md [hunk 1]
```diff
@@ -0,0 +1,1220 @@
...
+So you can send `all` 
+
+**public method Elasticsearch.Net.ClusterHealthRequestParameters.WaitForActiveShards** *Declaration changed (Breaking)*
+
+2.x: `public ClusterHealthRequestParameters WaitForActiveShards(long wait_for_active_shards)`  
+5.x: `public ClusterHealthRequestParameters WaitForActiveShards(string wait_for_active_shards)`  
+
+## Visibility changes
+
+These were types/methods/properties/constructors that were public but had nbno business being so.
```
Comment (gmarz) [inline docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md]: `nbno` => `no`

File: docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md [hunk 2]
```diff
@@ -0,0 +1,1220 @@
...
+**public property Elasticsearch.Net.Transport&lt;TConnectionSettings&gt;.PipelineProvider** *Visibility was changed from public to private (Breaking)*
+
+2.x: `public IRequestPipelineFactory PipelineProvider { get; }`  
+5.x: `private IRequestPipelineFactory PipelineProvider { get; }`  
+
+
+
+## Rename of API related methods
+
+Impact low, these have been renamed to match their Methid name equivalents
```
Comment (gmarz) [inline docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md]: Methid => Method

File: docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md [hunk 3]
```diff
@@ -0,0 +1,1220 @@
...
+
+**public method Elasticsearch.Net.IElasticLowLevelClient.TasksList&lt;T&gt;** *Declaration changed (Breaking)*
+
+2.x: `public ElasticsearchResponse<T> TasksList<T>(Func<TasksListRequestParameters, TasksListRequestParameters> requestParameters)`  
+5.x: `public ElasticsearchResponse<T> TasksList<T>(Func<ListTasksRequestParameters, ListTasksRequestParameters> requestParameters)`  
+
+
+## Rest spec updates
+
+These are breaking changes due to the elasticsearch 5.0 rest spec changing
```
Comment (gmarz) [inline docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md]: elasticsearch => Elasticsearch

File: docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md [hunk 4]
```diff
@@ -0,0 +1,1220 @@
...
+
+**public method Elasticsearch.Net.UpdateByQueryRequestParameters.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public UpdateByQueryRequestParameters RequestsPerSecond(Single requests_per_second)`  
+5.x: `public UpdateByQueryRequestParameters RequestsPerSecond(long requests_per_second)`  
+
+
+## Removed in 5.x after obsolete period
+
+These are types/properties/methods obsoleted in NEST 2.x that have now been removed.
```
Comment (gmarz) [inline docs/5.0-breaking-changes/elasticsearch-net-breaking-changes.md]: obsoleted => marked obsolete

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 5]
```diff
@@ -0,0 +1,3410 @@
+# Breaking Changes
+
+Oh my goodness, this looks like a lot of breaking changes! This is true but please take note that this list is very very extensive 
+It includes every single *binary* breaking change. In alot of cases these will not necessarily equate to compiler errors.
```
Comment (gmarz) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: alot => a lot

Comment (russcam) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: a lot

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 6]
```diff
@@ -0,0 +1,3410 @@
...
+2.x: `public TDescriptor AutoExpandReplicas(string AutoExpandReplicas)`  
+5.x: `public TDescriptor AutoExpandReplicas(AutoExpandReplicas autoExpandReplicas)`  
+
+## DslPrettyPrintVisitor methods are now virtual
+
+## Nest visitors should be bound to interface
+
+The visitors should be passed interfaces not concrete types see: https://github.com/elastic/elasticsearch-net/pull/2320
+
+## Deprecated queryies are now removed
```
Comment (russcam) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: Deprecated queries

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 7]
```diff
@@ -0,0 +1,3410 @@
...
+**public property Nest.BulkUpdateOperation&lt;TDocument, TPartialDocument&gt;.Script** *Declaration changed (Breaking)*
+
+2.x: `public string Script { get; set; }`  
+5.x: `public IScript Script { get; set; }`  
+
+**public property Nest.IUpdateRequest&lt;TDocument, TPartialDocument&gt;.Script** *Declaration changed (Breaking)*  
+**public property Nest.IBulkUpdateOperation&lt;TDocument, TPartialDocument&gt;.Script** *Declaration changed (Breaking)*  
+**public property Nest.UpdateRequest&lt;TDocument, TPartialDocument&gt;.Script** *Declaration changed (Breaking)*  
+
+In some cases the we abused IScript to send template queries this is now fixed
```
Comment (gmarz) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: the we => we

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 8]
```diff
@@ -0,0 +1,3410 @@
...
+5.x: `public Union<bool, ISourceFilter> Source { get; set; }`  
+
+**public property Nest.IInnerHits.Source** *Declaration changed (Breaking)*  
+**public property Nest.ISearchRequest.Source** *Declaration changed (Breaking)*  
+**public property Nest.ITopHitsAggregation.Source** *Declaration changed (Breaking)*  
+**public property Nest.SearchRequest.Source** *Declaration changed (Breaking)*  
+**public property Nest.SearchRequest&lt;T&gt;.Source** *Declaration changed (Breaking)*  
+**public property Nest.TopHitsAggregation.Source** *Declaration changed (Breaking)*  
+
+`Exclude` and `Include` are now plural on `ISourceFilter`
```
Comment (russcam) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: in line with the change in Elasticsearch 5.0

File: docs/5.0-breaking-changes/nest-breaking-changes.md (hunk 6 above)
Comment (gmarz) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: queryies => queries

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 9]
```diff
@@ -0,0 +1,3410 @@
...
+**public method Nest.ElasticClient.GetAlias** *Declaration changed (Breaking)*
+
+2.x: `public IGetAliasesResponse GetAlias(IGetAliasRequest request)`  
+5.x: `public IGetAliasResponse GetAlias(IGetAliasRequest request)`  
+
+**public method Nest.ElasticClient.GetAlias** *Declaration changed (Breaking)*  
+**public method Nest.IElasticClient.GetAlias** *Declaration changed (Breaking)*  
+**public method Nest.IElasticClient.GetAlias** *Declaration changed (Breaking)*  
+
+## Sugest is bound over T
```
Comment (russcam) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: Suggest

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 10]
```diff
@@ -0,0 +1,3410 @@
...
+**public property Nest.ITypeMapping.Meta** *Declaration changed (Breaking)*
+**public property Nest.PutMappingRequest.Meta** *Declaration changed (Breaking)*
+
+2.x: `public FluentDictionary<string, object> Meta { get; set; }`  
+5.x: `public IDictionary<string, object> Meta { get; set; }`  
+
+
+## Visibility changes
+
+Impact low, these are types/methods/constructors that were never suppose to be public.
```
Comment (gmarz) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: suppose => supposed

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 11]
```diff
@@ -0,0 +1,3410 @@
...
+**public property Nest.ISearchResponse&lt;T&gt;.TookAsLong** *Removed (Breaking)*  
+**public property Nest.ITermVectorsResponse.TookAsLong** *Removed (Breaking)*  
+**public property Nest.PercolateCountResponse.TookAsLong** *Removed (Breaking)*  
+**public property Nest.SearchResponse&lt;T&gt;.TookAsLong** *Removed (Breaking)*  
+**public property Nest.TermVectorsResponse.TookAsLong** *Removed (Breaking)*  
+
+
+## Allow source filter to send false
+
+In NEST 2.x we would always send Source.Exlcude as `_source: { exclude: [""] }` in 5.x the we use a union of `bool`
```
Comment (gmarz) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: Exlcude => Exclude

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 12]
```diff
@@ -0,0 +1,3410 @@
...
+5.x
+```csharp
+[JsonPropertyAttribute("actions")]
+[JsonConverterAttribute(Nest.ActionsJsonConverter)]
+public Actions Actions { get; internal set; }
+```
+
+# Read only data types on responses.
+
+Responses now favor `IReadOnlyDictionary` and `IReadOnlyCollection` which are initialized as empty.
```
Comment (russcam) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: Use the open generic type names `IReadOnlyDictionary<TKey, TValue>` and `IReadOnlyCollection<T>`

File: docs/5.0-breaking-changes/nest-breaking-changes.md (hunk 9 above)
Comment (gmarz) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: Sugest => Suggest

File: docs/5.0-breaking-changes/nest-breaking-changes.md [hunk 13]
```diff
@@ -0,0 +1,3410 @@
...
+**public property Nest.HotThreadInformation.Threads**  
+**public property Nest.IAnalyzeResponse.Tokens**  
+**public property Nest.IAuthenticateResponse.Roles**  
+**public property Nest.IBulkResponse.Items**  
+**public property Nest.ICatResponse&lt;TCatRecord&gt;.Records**  
+**public property Nest.IClusterPendingTasksResponse.Tasks**  
+**public property Nest.IClusterRerouteResponse.Explanations**  
+**public property Nest.IGetSnapshotResponse.Snapshots**  
+**public property Nest.IGraphExploreResponse.Connections**  
+**public property Nest.IGraphExploreResponse.Failures
```
Comment (russcam) [inline docs/5.0-breaking-changes/nest-breaking-changes.md]: > With NEST 2.x you had to pass cancellation tokens as part of the `RequestConfiguration` we now bumped these to the methods themselves

to

> With NEST 2.x async methods, a cancellation tokens could be passed as part of the `RequestConfiguration`. This was not very discoverable and so each async method can now accept an optional cancellation token as an argument, making the API more async idiomatic.

**Comment by Mpdreamz** [comment]
ported to `master`

**Pull request merged**